### ISSUE TITLE
Add a new event property to Subscription Management related Amplitude events

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#7262]
This ticket covers events specific to Subscription Management.

Note: The `Amplitude.updateDefaultPaymentMethod_*` events for Stripe cannot be recorded for PayPal (see [this comment](https://jira.mozilla.com/browse/FXA-2895?focusedCommentId=243696&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-243696) below). A new event will be created for that situation. This new event is handled in FXA-3213.

---

We currently record metrics related to the funnel for subscriptions. We should record when a user signs up with PayPal.

A primer on metrics in FxA can be found at [https://mozilla.github.io/ecosystem-platform/docs/fxa-engineering/fxa-metrics.](https://mozilla.github.io/ecosystem-platform/docs/fxa-engineering/fxa-metrics.)  The type of metrics in discussion for this issue is Amplitude.  There's a list of these events at [https://docs.google.com/spreadsheets/d/1ttgV60YeurrMhmdlCZ5WNBJXa-PcnRmrGgOEow71E1U/edit#gid=0.](https://docs.google.com/spreadsheets/d/1ttgV60YeurrMhmdlCZ5WNBJXa-PcnRmrGgOEow71E1U/edit#gid=0.)

Amplitude allows a dictionary of "event properties" to be attached to each event.  For events where the user has the option of using Stripe or PayPal, we should add an event property of "payment_provider" with the value "Stripe" or "PayPal".

An example of such an event is the "fxa_pay_setup - success" event.  That event is emitted at [https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/lib/apiClient.ts#L239:L242, where you can see the event properties are directly passed into the function call.](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/lib/apiClient.ts#L239:L242)

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2895)
┆Issue Number: FXA-2895

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

This depends on FXA-2806, as the existing set of `Amplitude.updateDefaultPaymentMethod` events currently come from submitting the `PaymentForm` for Stripe customers as opposed to clicking the "Change" button and the different behavior downstream of that for PayPal per the design spec ( https://www.figma.com/file/ejDMYaKpm5bsYL5YvhItur/Payment-Integration?node-id=859%3A294 ) (particularly for error cases).

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Unfortunately, we cannot simply record the equivalent of `Amplitude.updateDefaultPaymentMethod_*()` (where * is PENDING, FULFILLED or REJECTED), as PayPal works differently from Stripe in this regard in most cases (1 and 2 below).

First, some background: Per FXA-2806, there are 3 states a PayPal customer's payment method could be in when they go to Subscription Management:
 1. No payment errors. Billing agreement is valid.
 2. General funding source error (e.g. insufficient funds). Billing agreement is valid.
 3. Invalid billing agreement.

Specifically in cases 1 and 2, while a Stripe customer can update their credit card by submitting a form on the Subscription Management page, a PayPal customer would click the "Change" button and be taken to an external PayPal page without any information being sent back to us. AFAIK, there is no IPN notification for funding source changes, so we would not be able to tell from the fxa-auth-server side either. As a partial solution, we could implement a "Retry" button for outstanding invoices, but that is out of scope for the MVP.

Only in case 3, where we have to reauthorize payment, would we know that the funding source was changed successfully or not.

After a lengthy Slack conversation with Barry ( https://mozilla.slack.com/archives/CG0AJ6E77/p1616096159021900 ), we agreed that the best strategy is to recognize that Stripe and PayPal work differently, so we should handle this event differently. Specifically, we should create a new event that is recorded when the "Change" button is clicked. We pass `paymentProvider` this new event as well as to the existing `Amplitude.updateDefaultPaymentMethod_*()` events.

What this means is that:

  * The new event will see `paymentProvider` with values of both `'stripe'` and `'paypal'`, since clicking the "Change" button is what triggers the Stripe form to appear in the Stripe case.
  * The existing `Amplitude.updateDefaultPaymentMethod_*()` events will never have a value of `'paypal'`.

This new event should be added to the Subscription Platform metrics spreadsheet ( https://docs.google.com/spreadsheets/d/1ttgV60YeurrMhmdlCZ5WNBJXa-PcnRmrGgOEow71E1U/edit#gid=0 ) and to our Amplitude dashboard (CC [~byee@mozilla.com] ).

Due to the increased scope, I'm going to break out adding the new Amplitude event into a separate ticket. Only that event should be blocked by FXA-2806.
